exp/controllers/azuremangedcontrolplane_reconciler.go

```diff
@@ -98,14 +100,16 @@ func (r *azureManagedControlPlaneReconciler) Reconcile(ctx context.Context, scop
 // Delete reconciles all the services in pre determined order
 func (r *azureManagedControlPlaneReconciler) Delete(ctx context.Context, scope *scope.ManagedControlPlaneScope) error {
 	managedClusterSpec := &managedclusters.Spec{
-		Name:          scope.ControlPlane.Name,
-		ResourceGroup: scope.ControlPlane.Spec.ResourceGroup,
-		Location:      scope.ControlPlane.Spec.Location,
-		Tags:          scope.ControlPlane.Spec.AdditionalTags,
-		Version:       strings.TrimPrefix(scope.ControlPlane.Spec.Version, "v"),
-		NetworkPlugin: scope.ControlPlane.Spec.NetworkPlugin,
-		NetworkPolicy: scope.ControlPlane.Spec.NetworkPolicy,
-		SSHPublicKey:  scope.ControlPlane.Spec.SSHPublicKey,
+		Name:            scope.ControlPlane.Name,
+		ResourceGroup:   scope.ControlPlane.Spec.ResourceGroup,
+		Location:        scope.ControlPlane.Spec.Location,
+		Tags:            scope.ControlPlane.Spec.AdditionalTags,
+		Version:         strings.TrimPrefix(scope.ControlPlane.Spec.Version, "v"),
+		NetworkPlugin:   scope.ControlPlane.Spec.NetworkPlugin,
+		NetworkPolicy:   scope.ControlPlane.Spec.NetworkPolicy,
+		SSHPublicKey:    scope.ControlPlane.Spec.SSHPublicKey,
+		DNSServiceIP:    scope.ControlPlane.Spec.DNSServiceIP,
```

Comment:
I don't see us setting serviceCIDR? That seems like a bug from my PR 🙁

Reply:
I think you're good :smile: -- I believe the owner cluster object is passed via ManagedControlPlaneScope to the reconciler from the amcp controller [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/exp/controllers/azuremangedcontrolplane_controller.go#L112).  It looks like the reconciler is setting the serviceCIDR in the managedClusterSpec [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/exp/controllers/azuremangedcontrolplane_reconciler.go#L133).